### ISSUE TITLE
Gulp tasks are listed in duplicate

### ISSUE THREAD

**Issue by anseki** [Microsoft/vscode#1930]
According to the docs ["Autodetecting Gulp, Grunt and Jake Tasks"](https://code.visualstudio.com/docs/editor/tasks#_autodetecting-gulp-grunt-and-jake-tasks), **`gulpfile.js` at project root** is loaded and tasks in that are listed automatically.

`gulpfile.js`:

``` js
var gulp = require('gulp');
gulp.task('gulpfileTASK1', function() {});
gulp.task('gulpfileTASK2', function() {});
gulp.task('gulpfileTASK3', function() {});
```

VS Code Command `>Tasks: Run Task`:

```
gulpfileTASK1
gulpfileTASK2
gulpfileTASK3
```

This autodetecting is disabled (`gulpfile.js` is ignored) if tasks are defined in `.vscode/tasks.json`, right?

`tasks.json`:

``` json
{
  "version": "0.1.0",
  "command": "something",
  "isShellCommand": true,
  "tasks": [
    {
      "taskName": "tasksjsonTASK1",
      "args": []
    },
    {
      "taskName": "tasksjsonTASK2",
      "args": []
    },
    {
      "taskName": "tasksjsonTASK3",
      "args": []
    }
  ]
}
```

VS Code Command `>Tasks: Run Task` (tasks in `gulpfile.js` are not listed):

```
tasksjsonTASK1
tasksjsonTASK2
tasksjsonTASK3
```

It seems that behavior above is right.
But if `gulp` is specified to the `command` above (`"command": "gulp"`), tasks in both files are listed.

VS Code Command `>Tasks: Run Task`:

```
gulpfileTASK1
gulpfileTASK2
gulpfileTASK3
tasksjsonTASK1
tasksjsonTASK2
tasksjsonTASK3
```

Of course the tasks in `tasks.json` are gulp tasks because `command` is `gulp`. `gulpfileTASK*` tasks should not be listed.

When `--gulpfile` is specified, these tasks are listed even if `gulpfile.js` is another named and it is in outside of project root.

`tasks.json`:

``` json
{
  "version": "0.1.0",
  "command": "gulp",
  "isShellCommand": true,
  "args": [
    "--gulpfile", "PATH/TO/OUTSIDE/OF/PROJECT/ANOTHER_NAME.js"
  ],
  "tasks": [
    {
      "taskName": "tasksjsonTASK1",
      "args": []
    },
    {
      "taskName": "tasksjsonTASK2",
      "args": []
    },
    {
      "taskName": "tasksjsonTASK3",
      "args": []
    }
  ]
}
```

VS Code Command `>Tasks: Run Task`:

```
gulpfileTASK1
gulpfileTASK2
gulpfileTASK3
tasksjsonTASK1
tasksjsonTASK2
tasksjsonTASK3
```

It seems that the autodetecting is enabled always when `command` is `gulp` even if there is not `gulpfile.js` at project root.

**Comment by anseki** [comment]
Temporary solution:
If the tasks in both file have same name, this issue does not occur.

BTW, I want to use `tasks.json` to specify `args` for Gulp.

**Comment by dbaeumer** [comment]
This got addressed with the task version 2.0.0 and the new gulp autodetect and task types.
